### PR TITLE
Fix grep command in git hooks to find pragma once

### PR DESCRIPTION
**Pull request by mar-celine**

### CODE REVIEW

File: tools/Hooks/pre-commit.sh
Comment (kidder) [inline tools/Hooks/pre-commit.sh:203]: I am confused; why is this command not the same as the one in CheckFiles.sh
Also note the typo in the comment in that file

Comment (nilsdeppe) [reply tools/Hooks/pre-commit.sh:203]: @wthrowe  changed one of them and not the other, and he's the bash expert so we will need to wait for his answer

Comment (kidder) [reply tools/Hooks/pre-commit.sh:203]: There seems to be a lot of duplication in the two files; is there no way to avoid this?

Comment (wthrowe) [reply tools/Hooks/pre-commit.sh:203]: Sorry, I forgot to test this one.  Should be either `! grep -q  '^#pragma once$' "${file}"` or `! grep -x -q '#pragma once' "${file}"`, but I'm OK merging as is and I'll clean it up later.  (Avoid parens when you don't need them in bash, they create subshells which sometimes have surprising results.)

In terms of duplication, I just compared the scripts to see what the differences are.  The biggest things are the commit hook has to handle commits that don't touch files of a particular type (which is not necessary but would be harmless in CheckFiles.sh) and has logic in some (but not most) of its checks to only check modified lines.  The latter could probably be dropped without causing much trouble.  The hook looks for enable_if and files missing a final newline, which CheckFiles does not, but that's probably an oversight.  CheckFiles also looks for the iostream header, and we probably don't want to block that in individual commits, but it could be done as a special check in CheckFiles.

I'll look into merging the two sets of checks.

Comment (nilsdeppe) [reply tools/Hooks/pre-commit.sh:203]: Thanks for the detailed comment! enable_if was an intentional difference since we might want to use it in some backend code just so it looks more familiar to other devs. The final newline is definitely an oversight, and the iostream is intentional for the exact reason you gave. I think merging the files would be great if possible, and I don't think checking the entire file in the git hooks is bad since we have had them active from the very beginning and so it wouldn't result in anyone being "responsible" for cleaning up after other people.

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**
